src/ethereum/txUtils.ts

```diff
@@ -45,20 +76,28 @@ export namespace txUtils {
   /**
    * Wait until a transaction finishes by either being mined or failing
    * @param  {string} txId - Transaction id to watch
-   * @param  {number} [retriesOnEmpty] - Number of retries when a transaction status returns empty
+   * @param  {number} [retriesOnEmpty] - Number of retries when a transaction status returns empty, if not provided it will retry indefinitely
    * @return {Promise<object>} data - Current transaction data. See {@link txUtils#getTransaction}
    */
-  export async function waitForCompletion(txId: string, retriesOnEmpty?: number): Promise<any> {
-    const isDropped = await isTxDropped(txId, retriesOnEmpty)
-    if (isDropped) {
-      return { hash: txId, status: TRANSACTION_STATUS.failed, isDropped: true }
-    }
-
+  export async function waitForCompletion(
+    txId: string,
+    retriesOnEmpty?: number
+  ): Promise<MinedTransaction | DroppedTransaction> {
+    let retries = 0
     while (true) {
       const tx = await getTransaction(txId)
 
       if (tx && !isPending(tx) && tx.receipt) {
-        return tx
+        if (isFailure(tx)) {
+          return { ...tx, status: TRANSACTION_STATUS.failed }
+        } else {
+          return { ...tx, status: TRANSACTION_STATUS.confirmed }
+        }
+      }
+
+      retries++
```

Comment:
As we talked offline we should count retries to return dropped status only if tx is `null`